src/utils.ts

```diff
@@ -102,14 +96,16 @@ export const processWithOrdering = async <T>(
 
   const aggregateErrors = Object.values(unprocessedRecordsByListId)
     .map((record) => record.error)
+    .filter(Boolean)
     .flat();
 
-  if (aggregateErrors.length > 0 && rejectOnError) {
-    throw new AggregateError(aggregateErrors);
-  }
-
   return {
     hasUnprocessedRecords: aggregateErrors.length > 0,
     unprocessedRecords: unprocessedRecordsByListId,
+    throwOnUnprocessedRecords: () => {
```

Comment:
non-blocking

IMO, this API feels weird. It feels clearer to just duplicate this code in-line at the two call sites.

Reply:
I don't know, I understand why the part about returning the throwing function might feel weird. In another scenario we would either:
a) duplicate code
b) write a helper that does the same thing but just takes some arguments, like the unprocessed records list, export it, and call it in both places.

To me, none of those feel great either. This one is more like the zod-style where they let you create a schema and lazily call `parse` to actually do the validation (also with the option to do it safely).

Reply:
I think one normal way of throwing an error like this would be to do it from a parent function that calls the SQS, DynamoDB, or Kinesis processing code, then checks the output to throw an error if needed. The processing code for the different queue providers would all have to implement a common interface though (seems like they already are mostly). Not a big deal.